### DISCUSSION TITLE
Use ELSA with an existing DB Context

### DISCUSSION THREAD

**Discussion by pdevito3** [elsa-workflows/elsa-core#3707] [Ideas]
## Current State
Currently, i can do something like this: 

```csharp
        services.AddElsa(options => options.UseEntityFrameworkPersistence(ef => ef.UsePostgreSql(currentConnectionString), false));
```

but under the hood it will add a new `ElsaDbContext` and require it to be used. 

## Desired state
Ideally, I'd like to only have one Dbcontext and just add the ELSA tables to it. Maybe like this:

```csharp
        services.AddElsa(options => options.UseEntityFrameworkPersistence<MyProjectDbContext>());
```

Maybe I could even add a couple settings:

```csharp
    services.AddElsa(options => options.UseEntityFrameworkPersistence<MyProjectDbContext>(new StorageOptions
    {
        SchemaName = "elsa",
        AutoRunMigrations = false
    }));
```

**Comment by sfmskywalker** [comment]
I think that should be doable.

**Reply by pdevito3** [reply]
should i create a ticket for this?

**Reply by sfmskywalker** [reply]
That would be great. And if you want, please feel free to submit a PR.
